### ISSUE TITLE
Missing JSON in body when importing from Swagger

### ISSUE THREAD

**Issue by DaRolla** [a85/POSTMan-Chrome-Extension#2617]
Hi there,

1) I use http://editor.swagger.io and choose File -> Open Example... -> petstore_full.yaml. 
2) Then I choose File -> Download YAML.
3) In Postman I choose Import and drag the downloaded YAML file

Right now there are several Folder with the Requests.

For example in the folder "pets" there is the post request "Add a new pet to the store". But opening it and choosing "Body" I only can read "Pet object that needs to be added to the store". In fact I am missing JSON here. 

Using Swagger UI the same request is filles with this JSON:

{
  "id": 0,
  "category": {
    "id": 0,
    "name": "string"
  },
  "name": "doggie",
  "photoUrls": [
    "string"
  ],
  "tags": [
    {
      "id": 0,
      "name": "string"
    }
  ],
  "status": "available"
}

So can you please add the feature of generating example JSON as Swagger UI does?

Best regards,
DaRolla

**Comment by DaRolla** [comment]
Here's the petstore_full.yaml as ZIP file...

[petstore_full.zip](https://github.com/postmanlabs/postman-app-support/files/693699/petstore_full.zip)

**Comment by DaRolla** [comment]
Here's a short documentation how to use Swagger UI...

http://swagger.io/docs/#swagger-ui-documentation-29

**Comment by vegetableman** [comment]
@DaRolla Thanks for reporting this issue 👍 . Will get back to you with more details.

**Comment by vegetableman** [comment]
@DaRolla Tagging this a feature request. Will keep you posted on this thread.

**Comment by sutdisi** [comment]
+1 That would be awesome feature.

**Comment by sutdisi** [comment]
Any update on this?

**Comment by dperconti** [comment]
Having this issue as well

**Comment by szepczynski** [comment]
Problem still exists

**Comment by jayonsoftware** [comment]
Any update on this, its just a pain typing a json when we should be able to just import.

**Comment by johnddias** [comment]
Adding request for this to be addressed, makes Swagger import practically pointless if we have to manually add the request body.

**Comment by AbrahamGo** [comment]
Having this issue too. This issue has a big impact because makes useless "import" option from swagger. Highest priority please!!

**Comment by kartiknk** [comment]
Having this issue as well.  Using com.smoketurner dropwizard-swagger 1.1-1.1 
Pretty useless if we have to manually modify request for each and every test case

**Comment by prokhorovn** [comment]
Vote for this issue. Lack of possibility to import swagger-doc examples makes import from swagger useless, because the API shape is the only thing, which could be imported.

**Comment by dordal** [comment]
+1 on this. Would really like this to get fixed; we're using Postman to demo our API and this makes it a lot harder for users.

**Comment by ManojAshok** [comment]
+1   facing same issue   Swagger 2.0 JSON in Postman 5.4.0

**Comment by julianohsf** [comment]
+1 facing same issw Swagger 2.0 JSON in Postman 5.5.0

**Comment by peeeto** [comment]
+1

**Comment by juberjj** [comment]
The issue persists on 5.5.2

**Comment by pedro-lb** [comment]
Still having this issue.

**Comment by sreejith-ms** [comment]
Any updates??

**Comment by Giri-oss** [comment]
+1 Vote for this issue. without this Swagger import is not helpful.

**Comment by vaishrachan** [comment]
+1 Vote for this issue

**Comment by a85** [comment]
We have been working on a new converter. Check out the blog post here: http://blog.getpostman.com/2018/03/02/sync-your-specs-😎/

**Comment by rstav-gt** [comment]
Is the new converter powering the latest Postman version's import feature?

**Comment by vanbroup** [comment]
The new converter panics on swagger `binary` types.

**Comment by hadjsaid-taha** [comment]
+1

**Comment by rodriguise** [comment]
New convert dies on complex specs.

**Comment by abhijitkane** [comment]
@rodriguise Can you provide a sample JSON/YAML which the new converter crashes on? We're looking to integrate this with Postman soon.

**Comment by davidwzuora** [comment]
@abhijitkane I've been testing out the new converter (`swagger2-postman2`). The converter isn't picking up the request body samples that are included in the Swagger spec; instead it's generating request body samples that aren't very meaningful.

Here's the test spec I'm using:
https://github.com/davidwzuora/swagger2-postman2/blob/origin-test/zuora-spec01.json

There's a request body sample on [line 76](https://github.com/davidwzuora/swagger2-postman2/blob/origin-test/zuora-spec01.json#L76):

```
      "example": {
        "baseObject": "Invoice",
        "condition": "changeType == 'UPDATE' && Invoice.Status == 'Posted' && Invoice.Status_old != 'Posted' && Invoice.Amount > 1000"
      }
```

But the [generated Postman collection](https://github.com/davidwzuora/swagger2-postman2/blob/origin-test/zuora-postman01.json#L39) has an auto-mocked sample:

```
          "raw": "{\n  \"baseObject\": \"eu su\",\n  \"condition\": \"Lorem\"\n}"
```

I'm using the latest version of `swagger2-postman2` from GitHub, with this converter script:
https://github.com/davidwzuora/swagger2-postman2/blob/origin-test/zuora-convert01.js

It would be awesome if `swagger2-postman2` could pick up the request samples that are included in the Swagger spec, especially if you're going to integrate the converter into Postman.

Please let me know if you need any more info or test cases. Thanks!

**Comment by Tantalor93** [comment]
+1

**Comment by rodrigoimas** [comment]
+1

**Comment by antydemant** [comment]
+1

**Comment by benbward** [comment]
What is the status of this?  Does this screenshot reflect the same bug?
![image](https://user-images.githubusercontent.com/1945364/52360079-c9789400-2a00-11e9-94aa-994d46e4d6e7.png)

**Comment by sflusov** [comment]
Hi everyone,
I've find solution, you can use a next service (https://mermade.org.uk/openapi-converter) for converting Swagger document to document with the format OAS3 (Open Api Specification).

Of course, you can use this library directly (https://github.com/Mermade/oas-kit)

Let's have a result in a postman
![image](https://user-images.githubusercontent.com/11510074/52855999-3512dd80-3134-11e9-81b9-fcd07a41098d.png)

**Comment by ivanruslimcdohl** [comment]
facing the same problem

**Comment by tr1ck3r** [comment]
Hoping to see this fixed soon.  The Swagger 2.0 support claim is misleading with this defect.  Even the examples you link to from your docs reproduce this problem.  https://learning.getpostman.com/docs/postman/collections/data_formats/#importing-swagger

**Comment by silverbow25** [comment]
> Hi everyone,
> I've find solution, you can use a next service (https://mermade.org.uk/openapi-converter) for converting Swagger document to document with the format OAS3 (Open Api Specification).
> 
> Of course, you can use this library directly (https://github.com/Mermade/oas-kit)
> 
> Let's have a result in a postman
> ![image](https://user-images.githubusercontent.com/11510074/52855999-3512dd80-3134-11e9-81b9-fcd07a41098d.png)

This works.  Just make sure to collapse your file before importing it to the converter or it might be too big.  Once I had it converted it correctly imported into Postman.

**Comment by a85** [comment]
Everyone, this is part of a much larger initiative but we are introducing the ability to define APIs in Postman using schemas. You will be able to document schemas for both requests and responses using whatever format you typically use. Postman will then take this as another input in generating docs, building mocks, or adding tests. It looks like this: 

<img width="1552" alt="Screenshot 2019-04-30 14 02 13" src="https://user-images.githubusercontent.com/653409/56995066-8d635400-6b55-11e9-9720-0888d91e4e82.png">

The schema formats are converted to Postman Collections using converters that we have open-sourced and will be distributed through a plug-in mechanism separately.

Along with this, we will be introducing "annotations" for collections as well as schemas soon so you will be able to annotate parts of your request/responses.

We'd love your feedback!

**Comment by Gambler13** [comment]
@a85 
Sounds very interesting. Is this feature already published? I can't see it in v7.0.9

**Comment by a85** [comment]
This will be out in 7.1

**Comment by a85** [comment]
The feature is rolling out in production now in 7.1. Integration of schemas with generated docs is not done yet but we are working on it.

**Comment by schepil** [comment]
> Hi everyone,
> I've find solution, you can use a next service (https://mermade.org.uk/openapi-converter) for converting Swagger document to document with the format OAS3 (Open Api Specification).
> 
> Of course, you can use this library directly (https://github.com/Mermade/oas-kit)
> 
> Let's have a result in a postman
> ![image](https://user-images.githubusercontent.com/11510074/52855999-3512dd80-3134-11e9-81b9-fcd07a41098d.png)

Hi,
Imported successfully, but the body is still missing in collection ;(

**Comment by a85** [comment]
@warkrime Next on the list.

**Comment by mutaimwiti** [comment]
Any progress update on this?

**Comment by sky54521** [comment]
The latest version 7.2.0 still have this problem.

**Comment by mbratz** [comment]
I'm also looking forward to use this feature.

**Comment by abhijitkane** [comment]
@sky54521 @mbratz @mutaimwiti @DaRolla This should be fixed in Postman 7.2.1. In the petstore_full, example I can see the following request body being generated for the `Add a new pet to the store` request:

```
{
    "name": "<string>",
    "photoUrls": [
        "<string>",
        "<string>"
    ],
    "id": "<long>",
    "category": {
        "id": "<long>",
        "name": "<string>"
    },
    "tags": [
        {
            "id": "<long>",
            "name": "<string>"
        },
        {
            "id": "<long>",
            "name": "<string>"
        }
    ],
    "status": "<string>"
}
```

Any issues with the way the request body is generated can be opened as separate issues :). Closing this for now.

**Comment by rodrigoimas** [comment]
> @sky54521 @mbratz @mutaimwiti @DaRolla This should be fixed in Postman 7.2.1. In the petstore_full, example I can see the following request body being generated for the `Add a new pet to the store` request:
> 
> ```
> {
>     "name": "<string>",
>     "photoUrls": [
>         "<string>",
>         "<string>"
>     ],
>     "id": "<long>",
>     "category": {
>         "id": "<long>",
>         "name": "<string>"
>     },
>     "tags": [
>         {
>             "id": "<long>",
>             "name": "<string>"
>         },
>         {
>             "id": "<long>",
>             "name": "<string>"
>         }
>     ],
>     "status": "<string>"
> }
> ```
> 
> Any issues with the way the request body is generated can be opened as separate issues :). Closing this for now.

Great job, Postman Team! This has been a much anticipated feature!
